src/BizHawk.Emulation.Cores/Consoles/Nintendo/Gameboy/GambatteLink.IEmulator.cs

```diff
@@ -19,16 +21,26 @@ public bool FrameAdvance(IController controller, bool render, bool rendersound =
 
 			foreach (var s in GBLinkController.BoolButtons)
 			{
-				if (controller.IsPressed(s))
+				if (!controller.IsPressed(s)) continue;

```

Comment:
How much has this code been tested? Does it work as expected even with mix matching SGB vs non-SGB with multiple different combinations?

Reply:
I should say I didn't do much testing as I was rewriting it, only really checking for crashes.